src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -889,8 +909,8 @@ func (c *controller) UploadLogs(ctx context.Context, cancellog context.CancelFun
 				podName = pods[0].Name
 			}
 
-			if status.HasError() {
-				c.log.Infof("Error detected. Closing logs and aborting...")
+			if status.HasError() || status.Cancelled() {
```

Comment:
Maybe i forgot something but i don't remember why we have logic here? 
In any case i don't like the idea of adding new status. If this is needed somehow (i truly believe we can cancel logs from outside) i prefer to make status struct to handle cancel and error case together.

Reply:
why we can't just disable this statement 
https://github.com/openshift/assisted-installer/blob/master/src/main/assisted-installer-controller/assisted_installer_main.go#L84 ?

Reply:
After discussion I am implementing a mechanism to cancel all the go routines on cancel and error